openhtf/plugs/__init__.py

```diff
@@ -405,7 +414,8 @@ def initialize_plugs(self, plug_types=None):
         self.tear_down_plugs()
         raise
       self.update_plug(plug_type, plug_instance)
-    self._initialize_rpc_server()
+    if not partial:
```

Comment:
ugh this was already kind of gross when you got here.  how about `initialize_plugs()` always takes specific plugs to initialize, and there's a separate `initialize()` that calls `self.initialize_plugs(self._plug_types)` and then `self._initialize_rpc_server()`?